build-packages/check-public-api/check-public-api.spec.ts

```diff
@@ -1,52 +1,58 @@
 import mock from 'mock-fs';
-import { createLogger } from '@sap-cloud-sdk/util';
+import path from 'path';
+import * as core from '@actions/core';
 import {
   checkBarrelRecursive,
   checkIndexFileExists,
   exportAllInBarrel,
   parseBarrelFile,
   parseIndexFile,
-  parseTypeDefinitionFile,
+  parseExportedObjectsInFile,
   regexExportedIndex,
   typeDescriptorPaths
-} from './check-public-api';
+} from './index';
 
 describe('check-public-api', () => {
-  afterEach(() => mock.restore());
+  let errorSpy: any;
 
-  it('checkIndexFileExists fails if index file is not in root', () => {
-    mock({
-      root: {
-        dir1: {
-          'index.ts': ''
-        }
-      }
-    });
-    expect(() => checkIndexFileExists('root/index.ts')).toThrowError(
-      'No index.ts file found in root.'
-    );
+  beforeEach(() => {
+    errorSpy = jest.spyOn(core, 'error').mockImplementation();
+  });
+  afterEach(() => {
+    errorSpy.mockRestore();
+    mock.restore();
   });
 
   describe('exportAllInBarrel', () => {
-    afterEach(() => mock.restore());
+    it('checkIndexFileExists fails if index file is not in root', () => {
+      mock({
+        root: {
+          dir1: {
+            'index.ts': ''
+          }
+        }
+      });
+      checkIndexFileExists('root/index.ts');
+      expect(errorSpy).toHaveBeenCalledWith('No index.ts file found in root.');
+    });
 
     it('fails if internal.ts is not present in root', async () => {
       mock({
         src: {
           file1: '',
+          'internal.ts': '',
```

Comment:
After adding this line, the test fails in 18 (as it should) but still passes in 20/22. So I suspect the behavior is different. Maybe something to look into

Reply:
I've used a workaround with `accessSync` instead. Found a small section on [iojs.org](https://iojs.org/api/fs.html#fs_fs_access_path_mode_callback:~:text=This%20is%20useful%20for%20determining%20if%20a%20file%20exists) that suggests this as an alternative. 
Works well and all the tests pass as normal now.